File: aoc15/src/test/java/aoc15/day04/Day04Test.java
package aoc15.day04;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class Day04Test {

  @Test
  void part1Test() {
    var day = new Day04();

    assertEquals(254575, day.part1("bgvyzdsv"));
    assertEquals(609043, day.part1("abcdef"));
    assertEquals(1048970, day.part1("pqrstuv"));
  }
}

File: aoc25/src/test/java/day04/Day04Test.kt
package day04

import aoc15.loadExample
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day04Test {

    val filename = "Day04.txt"

    @Test
    fun part1Example() {
        val input = loadExample(filename)
        val subject = Day04()

        val result = subject.part1(input)

        assertThat(result).isEqualTo(13)
    }

    @Test
    fun part2Example() {
        val input = loadExample(filename)
        val subject = Day04()

        val result = subject.part2(input)

        assertThat(result).isEqualTo(43)
    }
}
